src/main/resources/avro/bdg.avdl

```diff
@@ -748,299 +748,304 @@ record Variant {
+  //union { null, boolean } downsampled = null;
 
-  /**
-   A string describing the provenance of this sample and the processing applied
-   in genotyping this sample.
-   */
-  union { null, string }  processingDescription = null;
+  // The Wilcoxon rank-sum test statistic of the base quality scores. The base quality
+  // scores are separated by whether or not the base supports the reference or the
+  // alternate allele.
+  //union { null, float } baseQRankSum = null;
 
-  /**
-   An array describing the genotype called at this site. The length of this
-   array is equal to the ploidy of the sample at this site. This array may
-   reference OTHER_ALT alleles if this site is multi-allelic in this sample.
-   */
-  array<GenotypeAllele> alleles = [];
+  // The Fisher's exact test score for the strand bias of the reference and alternate
+  // alleles.
+  //union { null, float } fisherStrandBiasPValue = null;
+
+  // The number of reads at this site with mapping quality equal to 0.
+  //union { null, int } mapq0Reads = null;
+
+  // The Wilcoxon rank-sum test statistic of the mapping quality scores.
+  //union { null, float } mqRankSum = null;
+
+  // The Wilcoxon rank-sum test statistic of the position of the base in the read at this site.
+  //union { null, float } readPositionRankSum = null;
+
+  // The log scale prior probabilities of the various genotype states at this site.
+  //array<float> genotypePriors = [];
+
+  // The log scaled posterior probabilities of the various genotype states at this site,
+  // in this sample.
+  //array<float> genotypePosteriors = [];  Does this map to GP in the VariantContextConverter?
+
+  // The log-odds ratio of being a true vs. false variant under a trained statistical model.
+  //union { null, float } vqslod = null;
+
+  // If known, the feature that contributed the most to this variant being classified as
+  // a false variant.
+  //union { null, string } culprit = null;
 
   /**
-   The expected dosage of the alternate allele in this sample.
+   Additional genotype attributes that do not fit into the standard fields above.
+   The values are stored as strings, even for flag, integer, and float types. VCF
+   genotype key values with Number=., Number=0, Number=1, and Number=[n] are shared across
+   all alternate alleles in the same VCF record. VCF genotype key values with Number=A are
+   split for multi-allelic sites into a single value. VCF genotype key values with Number=R
+   are split into an array of two values, [reference allele, alternate allele], separated
+   by commas, e.g. "0,1".
    */
-  union { null, float } expectedAlleleDosage = null;
+  map<string> attributes = {};
+}
+
+/**
+ Genotype.
+ */
+record Genotype {
 
   /**
-   The number of reads that show evidence for the reference at this site.
+   The variant for this genotype. May be left unset or set to null for performance reasons.
    */
-  union { null, int } referenceReadDepth = null;
+  union { null, Variant } variant = null;
```

Comment:
Flatten the core variant fields into genotype (`referenceAllele`, `alternateAllele`, etc), as we did with `contigName`, `start`, `end`.